### PR TITLE
Move towards native audio

### PR DESCRIPTION
**Pull request by masonschafercodes**
Here is what I have done so far:
- converted the existing sound files in `web/public/sounds` to the correct format
- created the `awc` file with the correct samples and sample rate for each sound
- created the `dat` file with all the sound types and soundset
- updated the `fxmanifest` to reflect the newly added files for the streamed audio

I figure there is a good chunk of work around the Lua code and some on the UI side to move towards using this instead of audio through NUI. Just opening this in case anyone was to work at it as well and to track the progress of it.

### CODE REVIEW

**Comment by thelindat** [comment]
> opening this in case anyone was to work at it as well and to track the progress of it.

Mark as draft?

**Comment by FjamZoo** [comment]
Reused my very hard written code and not even added me as a co-contributor I feel betrayed

**Comment by masonschafercodes** [comment]
> Reused my very hard written code and not even added me as a co-contributor I feel betrayed

sorry Zoo ♥️

**Review by LukeWasTakenn — COMMENTED** [review]
This would break all custom sounds people may have put into their sounds folder, perhaps a fallback to NUI playing sounds or automatic conversion would make sense?

**Comment by thelindat** [comment]
> perhaps a fallback to NUI playing sounds or automatic conversion would make sense

~~Ideally scrap NUI entirely but I'd like to see a tool to convert files (somebody was already working on it).~~

File: audio/dlc_oxdoorlock/oxdoorlock.awc.xml [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+   <Chunks>
+    <Item>
+     <Type>peak</Type>
+    </Item>
+    <Item>
+     <Type>data</Type>
+    </Item>
+    <Item>
+     <Type>format</Type>
+     <Codec>PCM</Codec>
```
Comment (Ehbw) [inline audio/dlc_oxdoorlock/oxdoorlock.awc.xml]: `ADPCM` should be used over `PCM` for its smaller sizes with no noticeable lost in quality

Comment (masonschafercodes) [reply audio/dlc_oxdoorlock/oxdoorlock.awc.xml]: we only figured this out today you rat, but you right I will update it 🤪

Comment (FjamZoo) [reply audio/dlc_oxdoorlock/oxdoorlock.awc.xml]: > we only figured this out today you rat, but you right I will update it 🤪

Stop with the excuses and just do what you are told you minion

**Comment by thelindat** [comment]
> perhaps a fallback to NUI playing sounds or automatic conversion would make sense?

Let's keep the NUI there as fallback to keep things simple; I don't mind merging after that.

**Comment by masonschafercodes** [comment]
> Let's keep the NUI there as fallback to keep things simple; I don't mind merging after that.

Perhaps introducing a new convar could be a solution to this, something like `setr doorlock:nativeaudio [true|false]`? Let me know if this isn't what you want to do.

**Comment by thelindat** [comment]
> Perhaps introducing a new convar could be a solution to this, something like `setr doorlock:nativeaudio [true|false]`? Let me know if this isn't what you want to do.

I'd like to include native audio as the go-to, but that also means supporting people in converting their ogg files to native audio; and breaking the sounds until then.

Sure there's now a guide on it and people are working on a tool to make it quicker, but I'd rather not deal with complaints about "newer versions breaking audio".

**Comment by masonschafercodes** [comment]
I can also PR docs to mention the new convar and native audio support if need be.

**Pull request merged**